pages/_app.tsx

```diff
@@ -1,8 +1,42 @@
+import Router from "next/router";
 import { ThemeProvider, CSSReset } from "@chakra-ui/core";
 import { AppProps } from "next/app";
 
 import theme from "../src/theme";
 
+import * as firebase from "firebase/app";
+import "firebase/analytics";
+
+const firebaseConfig = {
+  apiKey: process.env.apiKey,
+  authDomain: process.env.authDomain,
+  databaseURL: process.env.databaseURL,
+  projectId: process.env.projectId,
+  storageBucket: process.env.storageBucket,
+  messagingSenderId: process.env.messagingSenderId,
+  appId: process.env.appId,
+  measurementId: process.env.measurementId,
+};
+
+if (typeof window !== "undefined" && !firebase.apps.length) {
+  firebase.initializeApp(firebaseConfig);
+
+  if (firebaseConfig.measurementId) {
+    firebase.analytics.isSupported().then((isSupported) => {
+      if (isSupported) {
+        const analytics = firebase.analytics();
+
+        Router.events.on("routeChangeComplete", (url) =>
+          analytics.logEvent(`routeChangeComplete: ${url}`)
+        );
+        Router.events.on("hashChangeComplete", (url) =>
+          analytics.logEvent(`hashChangeComplete: ${url}`)
+        );
```

Comment:
Yep! It's not so useful now when it's just a splash page but the extra info is cool to know.

Also I have a feeling I might be doing it wrong because only the default events are showing up on the analytics dashboard on Firebase rn :laughing: